src/main/java/edu/njit/jerse/daikonplusplus/inv/Invariant.java

```diff
@@ -0,0 +1,64 @@
+package edu.njit.jerse.daikonplusplus.inv;
+
+import java.util.Objects;
+
+/**
+ * Represents a candidate or inferred invariant as a Java-style expression string, along with its
+ * validation status.
+ */
+public class Invariant {
+
+  /** The actual invariant expression, e.g., "x > 0". */
+  private final String expression;
+
+  /** Whether this invariant has been falsified by dynamic evidence. */
+  private boolean falsified;
+
+  /**
+   * Constructs a new invariant with the given expression. By default, it is assumed to be valid
+   * (not falsified).
```

Comment:
> By default, it is assumed to be valid (not falsified).

I wouldn't use the word "valid" for this - I would expect the word "valid" to mean something like "well-formed" in this context (i.e., that `expression` is actually a valid Java expression string). The name "falsified" implies that the variable starts off as false, anyway, so I don't think you need to say this explicitly.